### PR TITLE
GP2-2574 WTE Add to shortlist

### PR DESCRIPTION
**Pull request by bobmeredith**
Adds checkboxes (styled as hearts) to add/remove markets to/from 'basket'.
also, reshuffle of row headings to be in three columns to clean up for screen readers.

Mobile design tweaked up. In order to re-use the code from the table (which has the header comprising multiple columns) the mobile styling is a little clunky as it's forced to use a header-less table and make the column widths a fixed %, but this seemed better than any other options.

The issues seen in manual testing were due to the checkbox not being 'controlled'. Should be good now.

### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/GP2-2574
- [x] Jira ticket has the correct status.
- [x] [Changelog](CHANGELOG.md) entry added.

### Reviewing help

- [ ] Explains how to test locally, including how to set up appropriate data
- [x] Includes screenshot(s) - ideally before and after, but at least after

### Housekeeping

- [ ] Added all new environment variables to Vault.
- [ ] Cleaned up old feature flags
- [ ] Upgraded any vulnerable dependencies.
- [ ] Ran the `make manage download_geolocation_data` command
- [ ] I have updated security dependencies
- [ ] Python requirements have been re-compiled.
- [x] Frontend assets have been re-compiled.

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

**Comment by bobmeredith** [comment]
image

File: react-components/src/components/CompareMarkets/DataTable.jsx [hunk 1]
```diff
@@ -169,7 +178,11 @@ export default function DataTable(props) {
     )
   }
 
-  setBaseYear(cache[datasetName], comparisonMarkets, config)
+  const addRemoveShortlist = (market, add) => {
+    ;[addMarketItem, removeMarketItem][add ? 0 : 1](market)
```
Comment (froddd) [inline react-components/src/components/CompareMarkets/DataTable.jsx:182]: Not sure this is very readable, would this work?

```suggestion
    (add ? addMarketItem : removeMarketItem)(market)
```

Or maybe the slightly more verbose, but definitely clearer:
```suggestion
    add ? addMarketItem(market) : removeMarketItem(market)
```

**Review by froddd — COMMENTED** [review]
Code looks good, but some issues in testing

**Comment by bobmeredith** [comment]
image

File: react-components/src/components/CompareMarkets/blocks.jsx [hunk 2]
```diff
@@ -97,34 +97,70 @@ const renderColumnHeader = (columnConfig, context, mobile) => {
...
+const renderCountryRowHeader = ({
+  market,
+  removeMarket,
+  config,
+  selectedMarkets,
+  addRemoveShortlist,
+  mobile,
+}) => {
   // A row header in normal or mobile mode is the country label. In mobile mode there is no 'remove' button
+  const iso = market.country_iso2_code
```
Comment (froddd) [inline react-components/src/components/CompareMarkets/blocks.jsx:109]: Is this comment still relevant?

**Comment by froddd** [comment]
Only one comment: the mobile/tablet list of countries (screenshot above) isn't tabular data, it's a list with actions, and as such shouldn't be represented in a table in order to be accessible.

**Review by froddd — APPROVED** [review]
Perfect!

**Pull request merged**